packages/contracts-bedrock/src/L2/L1Block.sol

```diff
@@ -73,4 +85,38 @@ contract L1Block is ISemver {
         l1FeeOverhead = _l1FeeOverhead;
         l1FeeScalar = _l1FeeScalar;
     }
+
+    /// @notice Updates the L1 block values for a post-blob activated chain.
+    /// Params are passed in as part of msg.data in order to compress the calldata.
+    /// Params should be passed in in the following order:
+    ///   1. _basefeeScalar      L1 basefee scalar
+    ///   2. _blobBasefeeScalar  L1 blob basefee scalar
+    ///   3. _sequenceNumber     Number of L2 blocks since epoch start.
+    ///   4. _number             L1 blocknumber.
+    ///   5. _timestamp          L1 timestamp.
+    ///   6. _basefee            L1 basefee.
+    ///   7. _blobBasefee        L1 blobBasefee.
+    ///   8. _hash               L1 blockhash.
+    ///   9. _batcherHash        Versioned hash to authenticate batcher by.
+    function setL1BlockValuesEcotone() external {
+	    //require(msg.sender == DEPOSITOR_ACCOUNT, "L1Block: only the depositor account can set L1 block values");
+	    assembly {
+            // Revert if the caller is not the depositor account.
+            if xor(caller(), DEPOSITOR_ACCOUNT) {
+                mstore(0x00, 0x3cc50b45)
```

Comment:
This looks like it now returns a 4 byte selector - can you document what the error type is? The error type should also be defined and then tested that it returns properly in the revert test

Reply:
this was the code suggested by @clabby  :) I don't actually know what the type is, assumed perhaps the function selector?

Reply:
It is the 4 byte selector for an error of signature `NotDepositor()`.

```
cast sig "NotDepositor()"
```

Reply:
```
cast 4b 3cc50b45
NotDepositor()
```

:)

Reply:
added comments to explain this for solidity newbs like me

Reply:
..and made the test confirm the return value is as expected on revert.